docs/100-upgrade/migration-guides/3.4-3.5.mdx

```diff
@@ -0,0 +1,81 @@
+description:
+  This page lists the highlights for upgrading a project from Front-Commerce 3.4
+  to 3.5
+---
+
+<p>{frontMatter.description}</p>
+
+## Update dependencies
+
+Update all your `@front-commerce/*` dependencies to this version:
+
+```shell
+pnpm update "@front-commerce/*@3.5.0"
+```
+
+## Configuration updates
+
+In this new release, we've introduced some changes on how some configurations
+are set.
+
+### Cache
+
+We've introduce new cache related to the `config` object in `FrontCommerceApp`.
+These are now defined with a config provider, it allow you to change the
+behaviour of the configuration computing.
+
+| Config path                     | Env Variable                                |
+| ------------------------------- | ------------------------------------------- |
+| `config.cache.apiToken`         | `FRONT_COMMERCE_CACHE_API_TOKEN`            |
+| `config.cache.ignoreCacheRegex` | `FRONT_COMMERCE_BACKEND_IGNORE_CACHE_REGEX` |
+
+### Magento 1/2
+
+We have updated the way we use the Magento 1/2 configuration. It's now possible
+to access Magento OAuth configuration with:
+
+| Config path                              | Env Variable                                                                                   |
+| ---------------------------------------- | ---------------------------------------------------------------------------------------------- |
+| `config.magento.oauth.accesToken`        | `FRONT_COMMERCE_MAGENTO1_ACCESS_TOKEN` OR `FRONT_COMMERCE_MAGENTO2_ACCESS_TOKEN`               |
+| `config.magento.oauth.accessTokenSecret` | `FRONT_COMMERCE_MAGENTO1_ACCESS_TOKEN_SECRET` OR `FRONT_COMMERCE_MAGENTO2_ACCESS_TOKEN_SECRET` |
+| `config.magento.oauth.consumerKey`       | `FRONT_COMMERCE_MAGENTO1_CONSUMER_KEY` OR `FRONT_COMMERCE_MAGENTO2_CONSUMER_KEY`               |
+| `config.magento.oauth.consumerSecret`    | `FRONT_COMMERCE_MAGENTO1_CONSUMER_SECRET` OR `FRONT_COMMERCE_MAGENTO2_CONSUMER_SECRET`         |
+
+### Image Resizer
+
+Display of dev error message is now handled with a configProvider and the
+configuration can be accessible like this:
+
+| Config path                           | Env Variable                             |
+| ------------------------------------- | ---------------------------------------- |
+| `config.resizedImage.disableDevError` | `FRONT_COMMERCE_DEV_IMAGE_ERROR_DISABLE` |
+
+## Deprecation of env vars
+
+These env vars support have been removed of core code in favor of
+configurations:
+
+- `FRONT_COMMERCE_MAINTENANCE_MODE_FORCE_ENABLED`
+- `FRONT_COMMERCE_MAINTENANCE_MODE_AUTHORIZED_IPS`
+- `FRONT_COMMERCE_CONTRIBUTION_MODE_FORCE`
+
+To configure these, you now should use configuration files:
+
+- [`app/config/contributionMode.ts`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/skeleton/app/config/contributionMode.ts)
+- [`app/config/maintenance.ts`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/skeleton/app/config/maintenance.ts)
```

Comment:
We can either add a link to the UserConfig type, giving links to the configuration files we are using act also as examples

Reply:
Well my point is, from that config files, it's not clear that they belong in `front-commece.config.ts` so it would be preferable to rather do it in an example inline

instead of linking to 3 different files to show how it should be used.

The typings are irrelavant because the defineConfig has the typings.

suggestion:

```ts
import { defineConfig } from "@front-commerce/core/config";

export default defineConfig({
  maintenance: {
    force: process.env.FRONT_COMMERCE_MAINTENANCE_MODE_FORCE_ENABLED === "true",
    authorizedIps: process.env.FRONT_COMMERCE_MAINTENANCE_MODE_AUTHORIZED_IPS?.split?.(",") ?? []
  },
  contributionMode: {
     force: process.env.FRONT_COMMERCE_CONTRIBUTION_MODE_FORCE === "true",
  },
});
```

Reply:
To prove my point, just to post this example, I had to follow 3 links instead of just coping existing code from the docs.

Reply:
The above suggestion is if you want to show an example, otherwise keep it short and link to their defined types, and mention that it should be in the `front-commerce.config.ts::maintenace` and `front-commerce.config.ts::contributionMode` config.

Reply:
added your suggestion